packages/cli/src/cmds/validator/keymanager/decryptKeystores/threadPool.ts

```diff
@@ -1,7 +1,12 @@
+import path from "node:path";
 import {spawn, Pool, Worker, ModuleThread, QueuedTask} from "@chainsafe/threads";
 import {DecryptKeystoreArgs, DecryptKeystoreWorkerAPI} from "./types.js";
 import {maxPoolSize} from "./poolSize.js";
 
+// Worker constructor consider the path relative to the current working directory
+const workerDir =
+  process.env.NODE_ENV === "test" ? "../../../../../lib/cmds/validator/keymanager/decryptKeystores" : "./";
```

Comment:
why is this necessary?

Reply:
Mocha and Vitest handles the runtime dependencies differently. Mocha passes everything through ts-node where it translates internally to right compiled file. While vitest not ties to to load the worker with exact path given. If we don't do that it tries to load `./worker.js` which does not exits in that directory `./worker.ts` exists. 

So why we have to point the worker path to `lib` path in test env.

Reply:
Have you tried if `new URL('./worker.js', import.meta.url)` works?

Reply:
Yes tried that it didn't work. `import.meta.url` points to the current `src` directory.

Reply:
Can we make this a bit more robust at least right now it breaks if
- someone runs tests without NODE_ENV set (we should avoid on relying on this env to be set)
- the file is moved or folders are renamed (this will not be detected in a refactoring that moves files / folders

Reply:
@nazarhussain I noticed we are using this pattern already on unstable branch. This looks like a bug in vitest as this workaround is not reasonable in my opinion, as testing framework should not force you to use weird hacks in your src code.

I think what we should do is
- open a issue on vitest repo to get this resolved (if it doesn't exist already)
- create an issue on lodestar repository, we also should make sure to reference all lines that have this in the issue
https://github.com/ChainSafe/lodestar/blob/d3005bf58efb527d2887a922689d74e32358bf79/packages/beacon-node/src/chain/bls/multithread/index.ts#L33
https://github.com/ChainSafe/lodestar/blob/d3005bf58efb527d2887a922689d74e32358bf79/packages/beacon-node/src/network/core/networkCoreWorkerHandler.ts#L33

Reply:
I had created an issue for it. https://github.com/ChainSafe/lodestar/issues/6208

It's not a bug in vitest neither a hack. Mocha uses runtime `NODE_OPTIONS` to specify a node loader to `ts-node`, so every code that you run in Node had to go through `ts-node`, that's why it's slow. 

Vitest compile and deliver the code to test runner, on **runtime**, so why when it requests for a file `src/worker.js` it does not find it there.  

Will try to find a neat workaround to address this thing in separate PR. 

> @nazarhussain I noticed we are using this pattern already on unstable branch. This looks like a bug in vitest as this workaround is not reasonable in my opinion, as testing framework should not force you to use weird hacks in your src code.
> 
> I think what we should do is
> 
> * open a issue on vitest repo to get this resolved (if it doesn't exist already)
> * create an issue on lodestar repository, we also should make sure to reference all lines that have this in the issue
>   https://github.com/ChainSafe/lodestar/blob/d3005bf58efb527d2887a922689d74e32358bf79/packages/beacon-node/src/chain/bls/multithread/index.ts#L33
>   
>   https://github.com/ChainSafe/lodestar/blob/d3005bf58efb527d2887a922689d74e32358bf79/packages/beacon-node/src/network/core/networkCoreWorkerHandler.ts#L33

Reply:
> It's not a bug in vitest neither a hack.

Having to use `process.env.NODE_ENV === "test"` in your source code is definitely a hack and `../../../../lib/` as well, if there is no other way to make it work we have to bite the bullet but this is not pretty.

Reply:
The use of `process.env.NODE_ENV` to `production` or `test` is common practice in JS projects to optimize some logic for particular cases. That `path` looks ugly, because we are not using any import aliases.

Reply:
> The use of process.env.NODE_ENV to production or test is common practice in JS projects to optimize some logic for particular use cases

See [NODE ENV=production is a lie](https://www.youtube.com/watch?v=HMM7GJC5E2o), I don't think it good to rely on that. For testing it might be still ok, I would still consider it a hack though.

> That path looks ugly, because we are not using any import aliases.

My main concern is not that it is ugly but rather that it is really fragile and will break if any folder changes in that path.